src/RuleSet/DeclarationBlock.php

```diff
@@ -107,7 +132,9 @@ public static function parse(ParserState $parserState, ?CSSList $list = null): ?
             }
         }
         $result->setComments($comments);
-        RuleSet::parseRuleSet($parserState, $result);
+
+        RuleSet::parseRuleSet($parserState, $result->ruleSet);
```

Comment:
Even though this technically works, I'd find using the getter `getRuleSet()` more clean here.

Reply:
`parseRuleSet()` takes the `RuleSet` argument by reference, in order to fill it in with the the parsed `Rule`s, so the class property needs to be passed.

Reply:
As far as I understand it, `getRuleSet()` returns the instance, not a clone of it.

(PHP by default passes object kind-of-by-reference. I've explained this on page 17 of my [diploma thesis](https://github.com/oliverklee/pixy-thesis/blob/main/pixy-thesis.pdf) back then.)